core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1293,16 +1292,16 @@ class UnifiedLog(@volatile var logStartOffset: Long,
       } else {
         // We need to search the first segment whose largest timestamp is >= the target timestamp if there is one.
         if (remoteLogEnabled() && !isEmpty) {
-          if (remoteLogManager.isEmpty) {
+          if (remoteOffsetReader.isEmpty) {
```

Comment:
nit: 

Optional to reword the exception msg (or) skip it: 
```
AsyncOffsetReader is empty even though the remote log storage is enabled.
```

Reply:
I wonder if keeping "RemoteLogManager" is clearer as it's a term already visible, and known, by users.

Reply:
Ack.